### PR TITLE
Fix Ajax Session loss Feature for 401 Standard Conformity

### PR DESCRIPTION
**Pull request by nkenneweg**
This PR should be subject to discussion!
@meeee @cnrk @tessi

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -173,7 +173,12 @@ def require_login
       end
       respond_to do |format|
         format.any(:html, :atom) { redirect_to signin_path(:back_url => url) }
-        format.any(:xml, :js, :json)  { head :unauthorized, "Reason" => "login needed" }
+
+        if request.headers["X-ACCEPT-AUTH"] == "Session"
+          format.any(:xml, :js, :json)  { head :unauthorized, "Reason" => "login needed", 'WWW-Authenticate' => 'Session realm="OpenProject API"' }
+        else
+          format.any(:xml, :js, :json) { head :unauthorized, 'WWW-Authenticate' => 'Basic realm="OpenProject API"' }
+        end
```
Comment (mfrister) [inline app/controllers/application_controller.rb:194]: Assuming you didn't remove the "Reason" => "login needed" on purpose, I'd suggest this code

``` ruby
authentication_scheme = if request.headers["X-ACCEPT-AUTH"] == "Session"
  'Session' else 'Basic'
end
format.any(:xml, :js, :json)  { head :unauthorized,
                                "Reason" => "login needed",
                                'WWW-Authenticate' => authentication_scheme + ' realm="OpenProject API"' }
```

**Comment by mfrister** [comment]
I'd like to have a test for the new header.

Any reason you chose the all-caps `X-ACCEPT-AUTH` as header name? What about `X-Accept-Authentication` or `X-Authentication-Scheme`?

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (nkenneweg) [inline app/controllers/application_controller.rb:194]: You are right. I changed the code.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -173,7 +173,13 @@ def require_login
       end
       respond_to do |format|
         format.any(:html, :atom) { redirect_to signin_path(:back_url => url) }
-        format.any(:xml, :js, :json)  { head :unauthorized, "Reason" => "login needed" }
+
+        authentication_scheme = if request.headers["X-Authentication-Scheme"] == "Session"
+          'Session' else 'Basic'
+        end
```
Comment (hschink) [inline app/controllers/application_controller.rb]: You should correct the intendation:

```
authentication_scheme = if request.headers["X-Authentication-Scheme"] == "Session" 
                          'Session' else 'Basic'
                        end
```

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -173,7 +173,13 @@ def require_login
...
+        format.any(:xml, :js, :json)  { head :unauthorized,
+          "Reason" => "login needed",
+          'WWW-Authenticate' => authentication_scheme + ' realm="OpenProject API"' }
```
Comment (hschink) [inline app/controllers/application_controller.rb]: The same here...

**Comment by hschink** [comment]
Please merge **after** indentation is fixed and updating the changelog.

**Pull request merged**